### DISCUSSION TITLE
v0.6.2 Pre-release thread

### DISCUSSION THREAD

**Discussion by masenf** [reflex-dev/reflex#4057] [Release]
Please report any problems you encounter in this discussion thread or as separate issues on reflex-dev/reflex.

```console
pip install 'reflex~=0.6.2a'
```

# Schedule

2024-10-03 19:35 PDT: 0.6.2a1 pre-release published for testing
2024-10-07 13:20 PDT: 0.6.2a2 pre-release published
2024-10-08 09:30 PDT: Public release of 0.6.2

# Release Notes

## Changes in 0.6.2a2

* fail safely when pickling (#4085)
* Get default for backend var defined in mixin (#4060)
* catch CancelledError in lifespan hack for windows (#4083)
* convert literal type to its variants (#4062)

## Improvements

* All SVG elements are now supported
  * feat: Add support for missing SVGs by @ChinoUkaegbu in https://github.com/reflex-dev/reflex/pull/3962
* Event triggers can now use `rx.cond`
  * support eventspec/eventchain in var operations by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4038
* Faster state serialization by replacing `dill` with normal `pickle`
  * [ENG-3867] Garden Variety Pickle by @masenf in https://github.com/reflex-dev/reflex/pull/4054

## Bug Fixes

* Handle `primary_key` defined in `sa_column`
  * a friendly little helper by @Kastier1 in https://github.com/reflex-dev/reflex/pull/4021
* [ENG-3870] rx.call_script with f-string var produces incorrect code by @masenf in https://github.com/reflex-dev/reflex/pull/4039

## Dynamic Components

* Use an equality check instead of startswith by @masenf in https://github.com/reflex-dev/reflex/pull/4024
* add of_type to _evaluate by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4051
* bundle chakra in window for CSR by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4042
* Include emotion inside of dynamic components by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4052

## Error Messages

* improve granian error message when not installed by @Lendemor in https://github.com/reflex-dev/reflex/pull/4037
* [ENG-3476]Setting State Vars that are not defined should raise an error by @ElijahAhianyo in https://github.com/reflex-dev/reflex/pull/4007
* set loglevel to info with hosting cli by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4043
* default should be warning for subprocesses not info by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4049
* remove var operation error by @adhami3310 in https://github.com/reflex-dev/reflex/pull/4053

## Other Changes
* use pathlib as much as possible by @Lendemor in https://github.com/reflex-dev/reflex/pull/3967
* Track the last reflex run time by @ElijahAhianyo in https://github.com/reflex-dev/reflex/pull/4045
* move router dataclasses in their own file by @Lendemor in https://github.com/reflex-dev/reflex/pull/4044

## New Contributors
* @ChinoUkaegbu made their first contribution in https://github.com/reflex-dev/reflex/pull/3962
* @Kastier1 made their first contribution in https://github.com/reflex-dev/reflex/pull/4021

**Full Changelog**: https://github.com/reflex-dev/reflex/compare/v0.6.1...release/reflex-0.6.2

**Comment by bertbarabas** [comment]
v0.6.2a1 found a couple of UndefinedStateVarErrors! Easy for me to define those and keep going. So far my app works.  I wonder if the undefined vars could have been causing issues I didn't know about...

**Comment by masenf** [comment]
0.6.2 is released!
